src/domain/bitcoin/lightning/ln-invoice.ts

```diff
@@ -27,17 +27,19 @@ export const decodeInvoice = (
     ? decodedInvoice.cltv_delta
     : null
 
-  const routeHints: RouteHint[] = []
+  const routeHints: Hop[][] = []
   if (decodedInvoice.routes) {
-    decodedInvoice.routes.forEach((route) =>
-      routeHints.push({
-        baseFeeMTokens: route.base_fee_mtokens,
-        channel: route.channel,
-        cltvDelta: route.cltv_delta,
-        feeRate: route.fee_rate,
-        nodePubkey: route.public_key as Pubkey,
-      }),
-    )
+    decodedInvoice.routes.forEach((rawRoute) => {
```

Comment:
isn't this a reduce?

Reply:
I think it's more of a mapping... it's getting at the hop objects inside a `route` array which is collected inside a `routes` array, and remapping the property names (and casting types where appropriate).